File: packages/test/polyglot/python/channel.py
from __future__ import annotations

import json
import sys
from pathlib import Path

from dbx_tools.bus import channel_name


def main() -> None:
    fixture_path = Path(sys.argv[1] if len(sys.argv) > 1 else "fixtures/channel.json")
    cases = json.loads(fixture_path.read_text())
    results = [{"name": case["name"], "result": channel_name(case["input"])} for case in cases]
    print(json.dumps(results, separators=(",", ":")))


if __name__ == "__main__":
    main()

File: packages/test/polyglot/src/channel.ts
import { readFileSync } from "node:fs";

import { PostgresTopicBus } from "@dbx-tools/postgres";

export interface ChannelCase {
  name: string;
  input: unknown;
  expected: string;
}

export interface ChannelResult {
  name: string;
  result: string;
}

export function readChannelCases(path: string): ChannelCase[] {
  return JSON.parse(readFileSync(path, "utf8")) as ChannelCase[];
}

export function runTypeScriptChannelCases(cases: ChannelCase[]): ChannelResult[] {
  return cases.map(({ name, input }) => ({
    name,
    result: new PostgresTopicBus({} as never, { channel: input }).channelName,
  }));
}
